learning/tour-of-beam/frontend/lib/pages/tour/widgets/solution_button.dart

```diff
@@ -42,7 +43,23 @@ class SolutionButton extends StatelessWidget {
                 : null,
           ),
         ),
-        onPressed: tourNotifier.toggleShowingSolution,
+        onPressed: () {
+          // TODO(nausharipov): resolve the conflict with save user code
+          showDialog(
+            context: context,
+            builder: (context) => Dialog(
+              backgroundColor: Colors.transparent,
+              child: BeamAlertDialog(
```

Comment:
Duplicate code.
`BeamAlertDialog.show(context, ...);` ?

Reply:
Moved `Dialog` into BeamAlertDialog, because future users of the widget might not know about the 
```
Future<void> show(BuildContext context) async {
    await showDialog(
      context: context,
      builder: build,
    );
 }
```
function and use showDialog.